raiden/network/proxies/token_network.py

```diff
@@ -708,63 +678,128 @@ def set_total_deposit(
         """
         typecheck(total_deposit, int)
 
-        with self.channel_operations_lock[partner], self.deposit_lock:
-            previous_total_deposit = self._detail_participant(
-                channel_identifier=channel_identifier,
-                detail_for=self.node_address,
-                partner=partner,
-                block_identifier=given_block_identifier,
-            ).deposit
+        if total_deposit <= 0 and total_deposit >= UINT256_MAX:
+            msg = f"Total deposit {total_deposit} is not in range [1, {UINT256_MAX}]"
+            raise RaidenUnrecoverableError(msg)
 
-            token_address = self.token_address()
-            token = Token(
-                jsonrpc_client=self.client,
-                token_address=token_address,
-                contract_manager=self.contract_manager,
-            )
+        with self.channel_operations_lock[partner], self.deposit_lock:
```

Comment:
This is missing the acquire for the token lock. again, please double check for dead locks.

Reply:
The lock is acquired when `approve` / `balanceOf` are called.

Reply:
Changed... now the lock is acquired in token_network proxy